[DataGrid] Deselect row in non checkboxSelection grid (#1813)
mui-org/material-ui-x: docs/src/pages/components/data-grid/selection/selection.md

### Single row selection

> Single row selection is enable by default with the `DataGrid` component. While deselecting a row, hold the <kbd class="key">CTRL</kbd> key. For the `XGrid`, you need to disable multiple row selection with `disableMultipleSelection={true}`.

Reviewer: One line per sentence is easier to review, we aim for writing docs like this.

```suggestion
Single row selection is enable by default with the `DataGrid` component.
To unselect a row, hold the <kbd class="key">CTRL</kbd> key and click on it.
For the `XGrid`, you need to disable multiple row selection with `disableMultipleSelection={true}`.
```

### Single row selection

Single row selection is enable by default with the `DataGrid` component.

> To unselect a row, hold the <kbd class="key">CTRL</kbd> key and click on it.

Reviewer 2: We just have CTRL here. It's a bit confusing 😆

Reviewer: In https://github.com/mui-org/material-ui-x/pull/1813#discussion_r645833808, I have tried to argue why Shift is harmful to this feature.

Reviewer 3: Pressing <kbd>Shift</kbd> while clicking the rows selects the text too. It's better to not support this key.

![hNBz68IIRK](https://user-images.githubusercontent.com/42154031/121016339-659e3b80-c772-11eb-914a-ad3775968062.gif)

Reviewer: Actually, we could open a new GitHub issue for https://material-ui.com/components/data-grid/selection/#multiple-row-selection. Shouldn't <kbd>Shift</kbd> should allow range selection, as it does on Excel or on a Tree View?

Reviewer 3: There's https://github.com/mui-org/material-ui-x/issues/208 but given the benchmark examples it seems more of a cell range selection. But I totally agree to allow the selection of a range of rows with <kbd>Shift</kbd>.

Reviewer: Ok, I'm adding a screenshot to #208 to avoid confusion :). It's different

### Single row selection

Single row selection is enable by default with the `DataGrid` component.

> To unselect a row, hold the <kbd class="key">CTRL</kbd> or <kbd class="key">Shift</kbd> (use <kbd class="key">⌘ Command</kbd> on macOS) key and click on it.

Reviewer: I would propose we only mention CTRL per https://material-ui.com/components/data-grid/accessibility/#key-assignment-conventions
```suggestion
To unselect a row, hold the <kbd class="key">CTRL</kbd> or <kbd class="key">Shift</kbd> key and click on it.
```

It's more or less based on https://www.w3.org/TR/wai-aria-practices/#kbd_common_conventions.

Reviewer: I would vote against using <kbd>Shift</kbd>. You would end up unselecting the current row if you are in multi-select mode and slightly miss your click target. At least, macOS doesn't support it not 

```suggestion
To unselect a row, hold the <kbd class="key">CTRL</kbd> (use <kbd class="key">⌘ Command</kbd> on macOS) key and click on it.
```

Reviewer 3: Yes, <kbd>Shift</kbd> is generally used for extending the selection. It doesn't makes sense but `isMultipleKeyPressed` supports it. I'll open a follow-up PR to fix the helper.

Reviewer: `isMultipleKeyPressed` is only used for the multi-header sorting. Which is an interesting problem. on AG Grid only <kbd>Shift</kbd> is handled. But on handsontable, a click on the header selects the whole column, so they only implement the multi-sorting wth <kbd>Ctrl</kbd>. IHMO the isMultipleKeyPressed abstraction is wrong, it should be handled on case by case.

For the `XGrid`, you need to disable multiple row selection with `disableMultipleSelection={true}`.

{{"demo": "pages/components/data-grid/selection/SingleRowSelectionGrid.js", "bg": "inline"}}

### Multiple row selection [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-pro/)

> To select multiple rows, focus on the `XGrid` component and hold the <kbd class="key">CTRL</kbd> key while selecting rows.

Reviewer 2: Do we need the focus on the grid? I thought we fixed that 🤔

Author: Yes we fixed that 👍 . I rephrased it.

> To activate multiple selection, put focus on the `XGrid` component and hold the <kbd class="key">CTRL</kbd> key while selecting rows.

Reviewer 4: ```suggestion
To select multiple rows, focus the `XGrid` component and hold the <kbd class="key">CTRL</kbd> key while selecting rows.
```
(Cmd on Mac)

Reviewer 4: The shift key also works for selecting multiple rows one at a time, but should instead allow range selection, for example selecting row 2, holding shift, and selecting row 5 should also select rows 2, 3 & 4.

Author: > The shift key also works for selecting multiple rows one at a time, but should instead allow range selection, for example selecting row 2, holding shift, and selecting row 5 should also select rows 2, 3 & 4.

Yup, we can consider this too in a separate PR.
